StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory+FormSpec.swift

```diff
@@ -7,38 +7,111 @@
 //
 
 import Foundation
+import StripeCore
 @_spi(STP) import StripeUICore
 
 extension PaymentSheetFormFactory {
-
     func specFromJSONProvider(provider: FormSpecProvider = FormSpecProvider.shared) -> FormSpec? {
         guard let paymentMethodType = PaymentSheet.PaymentMethodType.string(from: paymentMethod) else {
             return nil
         }
         return provider.formSpec(for: paymentMethodType)
     }
 
-    func makeFormElementFromSpec(spec: FormSpec) -> FormElement {
+    func makeFormElementFromSpec(
+        spec: FormSpec,
+        additionalElements: [Element] = []
+    ) -> PaymentMethodElementWrapper<FormElement> {
         let elements = makeFormElements(from: spec)
-        return FormElement(autoSectioningElements: elements, theme: theme)
+        let formElement = FormElement(
+            autoSectioningElements: elements + additionalElements,
+            theme: theme)
+        return makeDefaultsApplierWrapper(for: formElement)
     }
 
     private func makeFormElements(from spec: FormSpec) -> [Element] {
+        // These fields may be added according to `configuration.billingDetailsCollectionConfiguration` if they
+        // aren't already present.
+        var billingDetailsFields: [FormSpec.PlaceholderSpec.PlaceholderField] = [
+            .name,
+            .email,
+            .phone,
+            .billingAddress,
+        ]
+
+        // These fields will need to be connected.
+        var countryElement: Element?
+        var billingAddressElement: Element?
+        var phoneElement: Element?
+
         var elements: [Element] = []
         for fieldSpec in spec.fields {
             if let element = fieldSpecToElement(fieldSpec: fieldSpec) {
+                // Check if this is a billing details field and process it.
+                switch fieldSpec {
+                case .name:
+                    billingDetailsFields.remove(.name)
+                case .email:
+                    billingDetailsFields.remove(.email)
+                case .billing_address:
+                    billingDetailsFields.remove(.billingAddress)
+                    billingAddressElement = element
+                case .country, .klarna_country:
+                    countryElement = element
+                case .placeholder(let placeholder):
+                    switch placeholder.field {
+                    case .name:
+                        billingDetailsFields.remove(.name)
+                    case .email:
+                        billingDetailsFields.remove(.email)
+                    case .phone:
+                        billingDetailsFields.remove(.phone)
+                        phoneElement = element
+                    case .billingAddress, .billingAddressWithoutCountry:
+                        billingDetailsFields.remove(.billingAddress)
+                        billingAddressElement = element
+                    default: break
+                    }
+                default: break
+                }
```

Comment:
Can we not simplify this? Something like:

```suggestion
        for fieldSpec in spec.fields {
            if let element = fieldSpecToElement(fieldSpec: fieldSpec) {
                if billingDetailsFields.contains(fieldSpec) {
                    billingDetailsFields.remove(fieldSpec)
                }
            
                // Check if this is a billing details field and process it.
                switch fieldSpec {
                case .billing_address:
                    billingAddressElement = element
                case .country, .klarna_country:
                    countryElement = element
                case .placeholder(let placeholder):
                    if billingDetailsFields.contains(fieldSpec) {
                        billingDetailsFields.remove(fieldSpec)
                    }
                    
                    switch placeholder.field {
                    case .phone
                        phoneElement = element
                    case .billingAddress, .billingAddressWithoutCountry:
                        billingAddressElement = element
                    default: break
                    }
                default: break
                }
```

Reply:
Not really
1. The array contains `PlaceholderSpec`s, not `FieldSpec`s, so they cannot be compared directly.
2. I need to check whether the FieldSpec exists either as a FieldSpec or a PlaceholderSpec, which makes it check the array twice.
3. I still need the switch to assign the elements accordingly, so the change would only remove two lines and IMO it is easier to follow this way.

Reply:
I don't follow points (2) and (3) since the example code I wrote is still doing both of those things, but point (1) makes sense, thanks for clarifying

Reply:
I don't follow point (3) since the example code I wrote is still doing that, but points (1) and (2) make sense, thanks for clarifying

Reply:
In point 3 I'm referring to still needing that switch, so in the end you are only removing 4 total cases, in LOCs the total difference is 3 lines, and IMO it is easier to follow doing what you need to do in each case, than removing the specs in two separate contains/remove than doing it on each case when needed.